File: GT/src/test/java/php/funccall_taint_propagation_sanitize_taint.php
<?php


function first($id)
{

    $sql = "SELECT username FROM users WHERE id =  $id";
    $sql = second($id, null);
    return mysql_query($sql);
}

function second($arg1, $arg2)
{
    $arg1 = sanitize($arg1);
    return $arg1;
}

File: YASAT/src/test/java/javaLang/funccallpropagationsanitize/Funccall_Taint_Propagation_Sanitize_Taint.java
package javaLang.funccallpropagationsanitize;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Funccall_Taint_Propagation_Sanitize_Taint {

    public void method(String id) throws SQLException {
        String sql = "SELECT username FROM users WHERE id=" + id;

        sql = second(sql, null);
        Connection conn1 = null;

        ResultSet res = conn1.createStatement().executeQuery(sql);
    }

    public String second(String id, String second) {
        return sanitize(id);
    }

    public String sanitize(String input) {
        return input;
    }
}